### ISSUE TITLE
TextBlock does not become white in a dark theme

### ISSUE THREAD

**Issue by cadilhac** [Kinnara/ModernWpf#169]
Hi,

I'm new to WPF and this library. Thanks for your help in advance.

I have set the app to use dark theme and it works quite well. For instance a checkbox or a slider will appear fine with the chosen accent color and a white text. But all my TextBlocks however stay black.
Is there something obvious I forgot?

**Comment by stevemonaco** [comment]
This feature is working here. Are you applying your own styles to the TextBox? If so, make sure your style specifies `BasedOn="{StaticResource DefaultTreeViewItemStyle}"` so it is based on the ModernWPF style and can dynamically theme.

**Comment by cadilhac** [comment]
> This feature is working here. Are you applying your own styles to the TextBox? If so, make sure your style specifies `BasedOn="{StaticResource DefaultTextBoxStyle}"` so it is based on the ModernWPF style and can dynamically theme.

My issue is about TextBlock, not TextBox. Still, in case this is only a typo from your part, I have paid attention not to have custom styles for TextBlock.

**Comment by stevemonaco** [comment]
> My issue is about TextBlock, not TextBox.

Oops. (TextBlock also works here, to confirm). Is there a way to reproduce the behavior you're experiencing? Or is there some pattern such as `TextBlock` being white when the child of a certain control? I am able to produce some of the effect if I forget to add `ui:WindowHelper.UseModernWindowStyle="True"` to the parent `Window`. (Using `ui:ThemeManager.RequestedTheme="Dark"` to force Dark Theme for the `Window`). Those are the most obvious potential missteps.

**Comment by cadilhac** [comment]
I have just <ui:ThemeResources RequestedTheme="Dark" AccentColor="#588528" /> set on the app. This is what makes another kind of control work (like checkbox).
Otherwise, I can't identify patterns where it works or not. It always fails.

**Comment by stevemonaco** [comment]
Setting `ui:WindowHelper.UseModernWindowStyle="True"` on each `<Window>` is necessary. Does the problem persist if you follow the [Quick Start](https://github.com/Kinnara/ModernWpf#quick-start) on a standalone project?

**Comment by cadilhac** [comment]
> Setting `ui:WindowHelper.UseModernWindowStyle="True"` on each `<Window>` is necessary. Does the problem persist if you follow the [Quick Start](https://github.com/Kinnara/ModernWpf#quick-start) on a standalone project?

I don't think this is necessary (it just adds decoration to the main window, like a modern title bar). If I remove it from the ModernWpf sample project, textblocks continue to work well under dark theme. And anyway, if I add it in my own project it does not solve my issue.

**Comment by cadilhac** [comment]
I have found what was missing:

`Foreground="{DynamicResource SystemControlPageTextBaseHighBrush}"`

It was in the wiki, under Common Tasks -> Styling the window.

It's a chance I tried it because the doc is very light and this code is in a section about styling the main window, without an explanation that it will impact some of the controls (attaching it to the main window did not work, I had to attach it to a Page). In the same code, there is also a use of IsThemeAware which does nothing in my case and there is no explanation either of what it does. Under the Controls section, there is just a screenshot for each control.  And what about this brush and all others? I think the manual can still be improved! ;)

Don't get me wrong, I'm not complaining. I'm very happy to see that this library exists and I thank the dev. This is just a statement about the state of the manual, as seen by a beginner.
